src/machine.rs

```diff
@@ -412,12 +412,17 @@ where
                 .map(|f| f(&instruction))
                 .unwrap_or(0);
             self.add_cycles(cycles);
-            if let Some(max_cycles) = self.max_cycles() {
-                if self.cycles() > max_cycles {
-                    return Err(Error::MaximumCyclesReached);
+            if let Some(available_cycles) = self.available_cycles() {
+                if self.cycles() > available_cycles {
+                    return Err(Error::IncorrectSpecifiedCycles);
                 }
             }
         }
+        if let Some(available_cycles) = self.available_cycles() {
+            if self.cycles() != available_cycles {
```

Comment:
Is this check redundant with line 416?

Reply:
So line 416 checks for infinite loop, while this can help check the case where the program takes 400 cycles actually, but we specify 500 cycles. They really serve different purposes.